ksql-cli/src/main/java/io/confluent/ksql/cli/RemoteCli.java

```diff
@@ -68,9 +69,16 @@ public void execute(String commandStrippedLine) throws IOException {
     });
   }
 
+  // Visible for testing
+  public boolean hasUserCredentials() {
+    return restClient.hasUserCredentials();
+  }
+
   private void validateClient() {
     try {
-      restClient.makeRootRequest();
+      RestResponse restResponse = restClient.makeRootRequest();
+      if (restResponse.isErroneous())
+        terminal.writer().format("Couldn't connect to the KSQL server: %s\n\n", restResponse.getErrorMessage().getMessage());
```

Comment:
I updated the `validateClient` to receive the `PrintWriter` as an argument. Most uses of terminal here and in the parent class are to access the `writer()`, so it may make sense to store a reference to it as a separate variable like `terminalWriter` in the parent.